tests/v2/integration/setup/main.go

```diff
@@ -1,51 +1,104 @@
 package main
 
 import (
+	"fmt"
+	"log"
+	"net"
 	"os"
 
 	"github.com/creasty/defaults"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/k3d"
 	rancherClient "github.com/rancher/rancher/tests/framework/clients/rancher"
 	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
 	"github.com/rancher/rancher/tests/framework/extensions/token"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
 	"github.com/sirupsen/logrus"
-	"gopkg.in/yaml.v2"
+)
+
+var (
+	agentTag         = os.Getenv("AGENT_TAG")
+	masterAgentImage = "rancher/rancher-agent:" + agentTag
+)
+
+const (
+	k3dClusterNameBasename = "k3d-cluster"
 )
 
 // setup for integration testing
 func main() {
 	rancherConfig := new(rancherClient.Config)
-	file := "config.yaml"
 
 	user := &management.User{
 		Username: "admin",
 		Password: "admin",
 	}
 
-	hostURL := "localhost:8443"
+	ipAddress := getOutboundIP()
+	hostURL := fmt.Sprintf("%s:8443", ipAddress.String())
 	token, err := token.GenerateUserToken(user, hostURL)
 	if err != nil {
 		logrus.Fatalf("error with generating admin token: %v", err)
 	}
 
+	clusterName := namegen.AppendRandomString(k3dClusterNameBasename)
+
 	cleanup := true
 	rancherConfig.AdminToken = token.Token
 	rancherConfig.Host = hostURL
 	rancherConfig.Cleanup = &cleanup
-	rancherConfig.ClusterName = "local"
+	rancherConfig.ClusterName = clusterName
 
 	if err := defaults.Set(rancherConfig); err != nil {
 		logrus.Fatalf("error with setting up config file: %v", err)
 	}
 
-	all := map[string]*rancherClient.Config{}
-	all[rancherClient.ConfigurationFileKey] = rancherConfig
-	yamlConfig, err := yaml.Marshal(all)
+	config.WriteConfig(rancherClient.ConfigurationFileKey, rancherConfig)
+
+	testSession := session.NewSession()
+
+	client, err := rancherClient.NewClient("", testSession)
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
+	}
+
+	agentSetting := &v3.Setting{}
+
+	agentSettingResp, err := client.Steve.SteveType("management.cattle.io.setting").ByID("agent-image")
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
```

Comment:
Just a nit about error logs, below are the same as well

Reply:
updated